### PR TITLE
Made ox, oy, cx, cy require units in Scala, Julia, and spec.

### PR DESCRIPTION
**Pull request by Ichoran**
Also ensured that they come in pairs.  Doesn't make much sense otherwise.

### CODE REVIEW

**Comment by Ichoran** [comment]
@MichaelCurrie - Can you review and merge if this seems reasonable?  Thanks.

File: src/julia/src/CommonWorm.jl [hunk 1]
```diff
@@ -115,10 +114,18 @@ function parsed_json_to_worm(d :: Dict{AbstractString, Any})
     end
     for i in 1:length(x)
         if length(x[i]) != length(y[i])
-            result = string("Failed to parse worm ", id, " lengths of x and y spines do not agree at t = ", t[i])
+            result = string("Failed to parse worm ", id, ": lengths of x and y spines do not agree at t = ", t[i])
             return result
         end
     end
+    if (haskey(d, "cx") != haskey(d, "cy"))
```
Comment (MichaelCurrie) [inline src/julia/src/CommonWorm.jl:121]: A clever if somewhat obfuscated way to do an XOR :)

The equivalent in Python:

`assert(not(('cx' in segment_keys) ^ ('cy' in segment_keys)))`

Also could have been written:

`assert(not(xor('cx' in segment_keys, 'cy' in segment_keys)))`

To be clear, I think your method is fine here, just found it interesting.

**Comment by MichaelCurrie** [comment]
Although the PR failed the Travis-CI Python tests, this is only because of errors introduced by a commit to master I made that should have been made to a branch I've now created, `python_speedup`.  So the changes in this PR are not breaking anything on the Python side (of course).

As for the spec, the line

> Units must be specified at least for "t", "x", and "y".

Seems like it should be 

> Units must be specified for all quantities used.  Since "t", "x", and "y" are required, this means units must be specified for at least "t", "x", and "y".

Otherwise it seems quite reasonable.

File: src/julia/src/CommonWorm.jl (hunk 1 above)
Comment (Ichoran) [inline src/julia/src/CommonWorm.jl:121]: Huh, I did that because I though it was the _most_ obvious!  (Also, because `^` means exponentiation in Julia.)

**Comment by Ichoran** [comment]
@MichaelCurrie - Can you check the updated wording?  We show an example with no data, which makes it a little confusing to merely say, "Specify units on the variables you use."

**Comment by MichaelCurrie** [comment]
Perfect, and your description of the degenerate case describes how the Python version handles it.

**Pull request merged**